src/plugins/plot/src/telemetry/PlotController.js

```diff
@@ -0,0 +1,240 @@
+        this.openmct = openmct;
+        this.objectService = objectService;
+        this.exportImageService = exportImageService;
+
+        $scope.pending = 0;
+
+        this.listenTo($scope, 'user:viewport:change:end', this.onUserViewportChangeEnd, this);
+        this.listenTo($scope, '$destroy', this.destroy, this);
+
+        this.config = this.getConfig(this.$scope.domainObject);
+        this.listenTo(this.config.series, 'add', this.addSeries, this);
+        this.listenTo(this.config.series, 'remove', this.removeSeries, this);
+        this.config.series.forEach(this.addSeries, this);
+
+        this.followTimeConductor();
+        window.plot = this;
+    }
+
+    eventHelpers.extend(PlotController.prototype);
+
+    PlotController.prototype.followTimeConductor = function () {
+        this.listenTo(this.openmct.time, 'bounds', this.updateDisplayBounds, this);
+        this.listenTo(this.openmct.time, 'timeSystem', this.onTimeSystemChange, this);
+        this.synchronized(true);
+    };
+
+    PlotController.prototype.loadSeriesData = function (series) {
+        this.startLoading();
+        var options = {
+            size: this.$element[0].offsetWidth,
+            domain: this.config.xAxis.get('key')
+        };
+
+        series.load(options)
+            .then(this.stopLoading.bind(this));
+    };
+
+    PlotController.prototype.addSeries = function (series) {
+        this.listenTo(series, 'change:yKey', function () {
+            this.loadSeriesData(series);
+        }, this);
+        this.loadSeriesData(series);
+    };
+
+    PlotController.prototype.removeSeries = function (plotSeries) {
+        this.stopListening(plotSeries);
+    };
+
+    PlotController.prototype.getConfig = function (domainObject) {
+        var configId = domainObject.getId();
+        var config = configStore.get(configId);
+        if (!config) {
+            var newDomainObject = domainObject.useCapability('adapter');
+            config = new PlotConfigurationModel({
+                id: configId,
+                domainObject: newDomainObject,
+                openmct: this.openmct
+            });
+            configStore.add(configId, config);
+        }
+        configStore.track(configId);
+        return config;
+    };
+
+    PlotController.prototype.onTimeSystemChange = function (timeSystem) {
+        this.config.xAxis.set('key', timeSystem.key);
+    };
+
+    PlotController.prototype.destroy = function () {
+        configStore.untrack(this.config.id);
+        this.stopListening();
+    };
+
+    PlotController.prototype.loadMoreData = function (range, purge) {
+        this.config.series.map(function (plotSeries) {
+            this.startLoading();
+            plotSeries.load({
+                    size: this.$element[0].offsetWidth,
+                    start: range.min,
+                    end: range.max
+                })
+                .then(this.stopLoading.bind(this));
+            if (purge) {
```

Comment:
This is to purge data outside the current visible range on time conductor adjustments.  I don't believe that is done elsewhere.  Hrm, I see some unused code related to `resetOnAppend` that I will remove.

Reply:
Ah, I see. I thought `.load` replaced data with the newly fetched data, which would be time bound by the server, but I see in [PlotSeries.fetch that it's merged with the existing data](https://github.com/nasa/openmct/blob/204f6be9d0ad7b82add248dc46f31efb72cf3592/src/plugins/plot/src/configuration/PlotSeries.js#L145). It seems as though in the case of a conductor bounds change, the merging (which includes sorting etc.) is redundant? The data could just be replaced with the requested telemetry returned by the Telemetry API?

Reply:
@akhenry theoretically, yes, it could be replaced completely with historical telem returned by the telem API.  However, if there is any delay between realtime feed and historical ingestion, or if the historical request takes long enough to return, then there could be a race resulting in a gap between historical and realtime streams.  I prefer to merge the data to prevent this from occurring.